pkg/expr/sql/frame_db_conv.go

```diff
@@ -137,275 +149,149 @@ func convertDataType(fieldType data.FieldType) mysql.Type {
 //
 //nolint:gocyclo
 func fieldValFromRowVal(fieldType data.FieldType, val interface{}) (interface{}, error) {
-	// the input val may be nil, it also may not be a pointer even if the fieldtype is a nullable pointer type
+	// if the input interface is nil, we can return an untyped nil
 	if val == nil {
 		return nil, nil
 	}
 
 	switch fieldType {
-	// ----------------------------
-	// Int8 / Nullable Int8
-	// ----------------------------
-	case data.FieldTypeInt8:
-		v, ok := val.(int8)
-		if !ok {
-			return nil, fmt.Errorf("unexpected value type for interface %v of type %T, expected int8", val, val)
-		}
-		return v, nil
-
-	case data.FieldTypeNullableInt8:
-		vP, ok := val.(*int8)
-		if ok {
-			return vP, nil
-		}
+	case data.FieldTypeInt8, data.FieldTypeNullableInt8:
 		v, ok := val.(int8)
 		if ok {
```

Comment:
I'm still new-ish to Go, but I feel like I've seen this idiom more often
```suggestion
		v, ok := val.(int8); if !ok {
```
ie. putting the guard clause first (to reduce code blocks/indentation). WDYT?

Same comment applies all the way down this file.

Reply:
That is the norm / more idiomatic way.

Although I moved it for two reasons:

One of the blocks has a two type cases in it, so error is at the bottom
The longer line is the error message, so that is less indented.